dcos/cosmos.py

```diff
@@ -0,0 +1,401 @@
+from six.moves import urllib
+
+from dcos import config, http, util
+from dcos.errors import (DCOSAuthenticationException,
+                         DCOSAuthorizationException,
+                         DCOSBadRequest,
+                         DCOSException,
+                         DCOSHTTPException)
+
+logger = util.get_logger(__name__)
+
+
+class Cosmos(object):
+    """A wrapper on cosmos that abstracts away http requests"""
+
+    def __init__(self, cosmos_url=None):
+        if cosmos_url is None:
+            self.cosmos_url = get_cosmos_url()
+        else:
+            self.cosmos_url = cosmos_url
+
+        self._http_method = {
+            'capabilities': 'get',
+            'package/add': 'post',
+            'package/describe': 'post',
+            'package/install': 'post',
+            'package/list': 'post',
+            'package/list-versions': 'post',
+            'package/render': 'post',
+            'package/repository/add': 'post',
+            'package/repository/delete': 'post',
+            'package/repository/list': 'post',
+            'package/search': 'post',
+            'package/uninstall': 'post',
+            'service/start': 'post'
+        }
+
+        self._request_versions = {
+            'capabilities': ['v1'],
+            'package/add': ['v1'],
+            'package/describe': ['v2', 'v1'],
+            'package/install': ['v2', 'v1'],
+            'package/list': ['v1'],
+            'package/list-versions': ['v1'],
+            'package/render': ['v1'],
+            'package/repository/add': ['v1'],
+            'package/repository/delete': ['v1'],
+            'package/repository/list': ['v1'],
+            'package/search': ['v1'],
+            'package/uninstall': ['v1'],
+            'service/start': ['v1']
+        }
+
+        self._special_content_types = {
```

Comment:
This aren't just for the experimental. This is for any content type type that doesn't conform to the usual format.